lib/ecto/adapters/sql.ex

```diff
@@ -1377,21 +1426,28 @@ defmodule Ecto.Adapters.SQL do
 
   @repo_modules [Ecto.Repo.Queryable, Ecto.Repo.Schema, Ecto.Repo.Transaction]
 
-  defp last_non_ecto([{mod, _, _, _} | _stacktrace], repo, last)
+  def last_non_ecto_stacktrace(size, stacktrace, repo) do
```

Comment:
Now that I think about it, since we will move the `Enum.reverse` inside this function (as it is an implementation detail of this function), it should be called `first_non_ecto_stacktrace`. And that makes sense, right? We want the first non-ecto stacktrace.